### ISSUE TITLE
Rule proposal: add an option for `restrict-plus-operands` to treat enum members as their underlying type

### ISSUE THREAD

**Issue by isiahmeadows** [typescript-eslint/typescript-eslint#4472]
### Description

I commonly use enums (especially `const enum`s) to represent constants in my code base, as they're useful for a number of reasons (ranging from namespacing for readability to the AOT evaluation offered to numeric constants).

I want to enforce this restriction (as it's very much a source of bugs), but I want enums to be treated as their underlying type for the sake of this because of how I use them.

### Fail

```ts
const enum Foo {
  Value,
}
let foo = "str" + Foo.Value;
```

### Pass

```ts
const enum Foo {
  Value,
}
let foo = 1 + Foo.Value;
```

**Comment by isiahmeadows** [comment]
I'll have to figure out what's causing the issue then. Because something somewhere is causing that issue to appear in my editor. I'll file a bug report once I narrow down the cause, but for now, I'll close this.
